pulp_rpm/app/fields.py

```diff
@@ -0,0 +1,57 @@
+from rest_framework import serializers
+from pulp_rpm.app.models import UpdateCollection, UpdateReference
+
+
+class UpdateCollectionPackageField(serializers.ListField):
+    """
+    A serializer field for the 'UpdateCollectionPackage' model.
+    """
+
+    def to_representation(self, value):
+        """
+        Get list of packages from UpdateCollections for UpdateRecord if any.
+
+        Args:
+            value ('pk' of `UpdateRecord` instance): UUID of UpdateRecord instance
+
+        Returns:
+            A list of dictionaries representing packages inside the collections of UpdateRecord
+
+        """
+        ret = []
+        collections = UpdateCollection.objects.filter(update_record=value)
+        for collection in collections:
+            ret.append({
+                'name': collection.name,
+                'shortname': collection.shortname,
+                'packages': collection.packages.values()
```

Comment:
This adds all the fields, including the internal one which should not be exposed.
Please check the model for UpdateCollectionPackage to see which fields should be exposed.

Reply:
refactored to show only reasonable fields